classes/misc/residual_sfx/residual_sfx.gd

```diff
@@ -24,10 +24,22 @@ func _init(sound: AudioStream, pos: Vector2):
 # Takes an existing AudioStreamPlayer(2D) and makes it act like a ResidualSFX
 # (outlive its parent, optionally destroy on finish).
 static func new_from_existing (
-	sfx,
+	sfx: Node,
 	scene_root: Node,
 	destroy_on_finished = true
 ):
+	# VALIDATE: Is the passed node an audio source?
+	assert(
```

Comment:
A crash seems excessive. An error should be fine, no? We're not handling saved data, nor would anything particularly serious occur should this be incorrect.